app/waf/src/controllers/adc.controller.ts

```diff
@@ -344,4 +347,24 @@ export class AdcController {
       }
     });
   }
+
+  private async cUserdata(): Promise<string> {
+    let root_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+    let admin_password: string = Math.random()
+      .toString(36)
+      .slice(-8);
+
+    const user_data_unencoded: string = `#cloud-config
+    runcmd:
+     - "echo \\"root:${root_password}\\" | chpasswd"
```

Comment:
From the second line on, the format is incorrect, there should be no indents. 
Just want to suggest you have the test well done before you request the review.

Reply:
At jikui second approval, it is testd with GUI first, then it is tested with `curl` CLI